vpr/src/draw/search_bar.cpp

```diff
@@ -336,34 +406,111 @@ void warning_dialog_box(const char* message) {
 void search_type_changed(GtkComboBox* self, ezgl::application* app) {
     auto type = gtk_combo_box_text_get_active_text(GTK_COMBO_BOX_TEXT(self));
     GtkEntry* searchBar = GTK_ENTRY(app->get_object("TextInput"));
+    GtkEntryCompletion* completion = GTK_ENTRY_COMPLETION(app->get_object("Completion"));
+    GtkTreeModel* netNames = GTK_TREE_MODEL(app->get_object("NetNames"));
+    GtkTreeModel* blockNames = GTK_TREE_MODEL(app->get_object("BlockNames"));
+    GtkWidget* keyLengthSpinButton = GTK_WIDGET(app->get_object("KeyLength"));
+    GtkWidget* keyLengthLabel = GTK_WIDGET(app->get_object("KeyLengthLabel"));
     //Ensuring a valid type was selected
     if (!type) return;
     if (type[0] == '\0') return;
     std::string searchType(type);
-    //Setting active completion model to blockCompleter if type selected is block Name
+
+    /*
+     * If search type is name, connecting search bar to completion,
```

Comment:
This might be a good place for a comment about key length and its performance impact (probably better than the earlier place I listed).